test/reports/encoding_reports/test_featureDistributionReport.py

```diff
@@ -0,0 +1,79 @@
+import os
+import random
+import shutil
+import string
+from unittest import TestCase
+
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+from immuneML.caching.CacheType import CacheType
+from immuneML.data_model.dataset.RepertoireDataset import RepertoireDataset
+from immuneML.data_model.encoded_data.EncodedData import EncodedData
+from immuneML.environment.Constants import Constants
+from immuneML.environment.EnvironmentSettings import EnvironmentSettings
+from immuneML.reports.ReportResult import ReportResult
+from immuneML.reports.encoding_reports.FeatureDistributionReport import FeatureDistributionReport
+
+
+class TestFeatureDistributionReport(TestCase):
+
+    def setUp(self) -> None:
+        os.environ[Constants.CACHE_TYPE] = CacheType.TEST.name
+
+    def _create_dummy_encoded_data(self, path):
+        n_subjects = 50
+        n_features = 30
+
+        kmers = [''.join(random.choices(string.ascii_uppercase, k=3)) for i in range(n_features)]
+
+        encoded_data = {
+            'examples': sparse.csr_matrix(
+                np.random.normal(50, 10, n_subjects * n_features).reshape((n_subjects, n_features))),
+            'example_ids': [''.join(random.choices(string.ascii_uppercase, k=4)) for i in range(n_subjects)],
+            'labels': {
+                "patient": np.array([i for i in range(n_subjects)])
+            },
+            'feature_names': kmers,
+            'feature_annotations': pd.DataFrame({
+                "sequence": kmers
+            }),
+            'encoding': "random"
+        }
+
+        dataset = RepertoireDataset(encoded_data=EncodedData(**encoded_data))
+
+        return dataset
+
+    def test_generate(self):
+        path = EnvironmentSettings.root_path / "test/tmp/featurevaluebarplot/"
```

Comment:
could you use a different name for the test folder? :)

Reply:
to elaborate; we already use "test/tmp/featurevaluebarplot/" for the featurevaluebarplot report, so if one or both of them fails they overwrite each other's results